entity-service-change-event-generator/src/main/java/org/hypertrace/entity/service/change/event/metric/EntityMetricsRegistry.java

```diff
@@ -0,0 +1,88 @@
+package org.hypertrace.entity.service.change.event.metric;
+
+import io.micrometer.core.instrument.Counter;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+import lombok.AllArgsConstructor;
+import org.hypertrace.core.serviceframework.metrics.PlatformMetricsRegistry;
+
+public class EntityMetricsRegistry {
+
+  private static final String ENTITIES_CREATE_COUNTER = "entities.create.counter";
+  private static final String ENTITIES_UPDATE_COUNTER = "entities.update.counter";
+  private static final String ENTITIES_DELETE_COUNTER = "entities.delete.counter";
+  private static final String ENTITY_TYPE_TAG = "entityType";
+  private static final String TENANT_ID_TAG = "tenantId";
+  private static final Map<EntityTypeTenantMetricPair, Counter> typeToTenantEntityCounter =
+      new ConcurrentHashMap<>();
+
+  private static final EntityMetricsRegistry ENTITY_METRICS_REGISTRY = new EntityMetricsRegistry();
+
+  // private constructor
+  private EntityMetricsRegistry() {}
+
+  public static EntityMetricsRegistry getEntityMetricsRegistry() {
+    return ENTITY_METRICS_REGISTRY;
+  }
+
+  public Counter getCreateCounter(String apiType, String tenantId) {
```

Comment:
An API type is not applicable to any entity except an API. If that's something that needs to be tracked, make the metrics api-specific - otherwise remove.